libflux/src/flux/semantic/types.rs

```diff
@@ -7,13 +7,41 @@ use std::{
     fmt,
 };
 
+// For use in generics where the specific type of map is not not mentioned.
+pub type SemanticMap<K, V> = BTreeMap<K, V>;
+pub type SemanticMapIter<'a, K, V> = std::collections::btree_map::Iter<'a, K, V>;
+
 #[derive(Debug, Clone)]
 pub struct PolyType {
     pub vars: Vec<Tvar>,
-    pub cons: HashMap<Tvar, Vec<Kind>>,
+    pub cons: TvarKinds,
     pub expr: MonoType,
 }
 
+pub type PolyTypeMap = SemanticMap<String, PolyType>;
+pub type PolyTypeMapMap = SemanticMap<String, SemanticMap<String, PolyType>>;
+
+#[macro_export]
+/// This is an redefinition of the maplit::hashmap macro. It allows us to alter
+/// the map type used.
+macro_rules! semantic_map {
```

Comment:
Can you give a little more detail here? Why do we need to redefine it?

Reply:
Oh I wanted to just do a passthrough to whichever one SemanticMap was aliased to, but couldn't figure it out, so I cloned the whole thing, then forgot about it.

Do you know how to just pass macro arguments to another macro?

Reply:
Oh I see, so we can update the type `SemanticMap` is aliasing without having to refactor the invocations of `semantic_map!`. I missed where you updated the map type in the body of the macro. I'm not quite sure what you mean by passthrough, but this seems like the correct approach to me.

Reply:
Was looking for an effective wrap of the macros, without providing the implementation. I had to read the macro docs more carefully, but I figured it out. It's a nice system, but some hard-to-read syntax at first.
```
macro_rules! aliased_map {
    ( $($x:tt)* ) => ( maplit::hashmap!( $( $x )* ) );
}
```

Reply:
That is saying match any tokens as the macro args and expand them as they are into the hash map! args.